apis/apiextensions/v1alpha1/composition_types.go

```diff
@@ -235,7 +234,10 @@ func (t *Transform) Transform(input interface{}) (interface{}, error) {
 	default:
 		return nil, errors.New(errTypeNotSupported(string(t.Type)))
 	}
-	if transformer == nil {
+	// An interface equals nil only if both the type and value are nil. Above,
+	// even if t.<Type> is nil, its type is assigned to "transformer" but we're
+	// interested in whether only the value is nil or not.
+	if reflect.ValueOf(transformer).IsNil() {
```

Comment:
This seems reasonable to me 👍 One other option I considered was expanding the `transformer` interface such that it had something like `IsValid()` that could do the underlying `nil` check (you can call functions on a nil pointer in go) and potentially any other validation needed for that specific transform. See an example here: https://play.golang.org/p/3h1vXVL0K4o

Reply:
I think in other possible places that these structs are used, the developers will need to do the `nil` check on the transform type they would like to use anyway since marking it as pointer implies that it can be empty. In this specific case, `nil` check on an interface needs some logic rather than the structs themselves. So, I'd prefer to keep that logic contained in this function where it's needed to keep the structs simple.

Reply:
@muvaf fair enough, looks like some linter errors to correct here

Reply:
just fixed, thanks!